From 42c1535e914d1afdf1222bc4d88732e5ea862ecf Mon Sep 17 00:00:00 2001
From: Mohammad Hasan Keramat J <ikeramat@protonmail.com>
Date: Wed, 30 Aug 2023 20:51:57 +0000
Subject: [PATCH] recovery: Disable `ask_to_ab_reboot` in user build

* in user build its not allowed to flash unsigned zip

Signed-off-by: Mohammad Hasan Keramat J <ikeramat@protonmail.com>
Change-Id: Iedcebc4aeaa1b1251619beeb3003bcc4e1bc9db9
---
 recovery.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/recovery.cpp b/recovery.cpp
index b5714d48..fcc03046 100644
--- a/recovery.cpp
+++ b/recovery.cpp
@@ -176,6 +176,9 @@ static bool yes_no(Device* device, const char* question1, const char* question2)
 }
 
 bool ask_to_ab_reboot(Device* device) {
+  if (get_build_type() == "user") {
+    return false;
+  }
   device->GetUI()->SetProgressType(RecoveryUI::EMPTY);
   return yes_no(device, "To install additional packages, you need to reboot recovery first",
                 "Do you want to reboot to recovery now?");
-- 
2.29.2

